ui/perfherder/compare/CompareTableView.jsx

```diff
@@ -62,7 +62,10 @@ export default class CompareTableView extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.location.search !== prevProps.location.search) {
+    if (
+      this.props.location.search !== prevProps.location.search ||
+      this.props.validated.newRevision !== prevProps.validated.newRevision
+    ) {
```

Comment:
This should only be checking if `this.props.validated` has changed because any changes to `this.props.location` is handled by the `withValidation` higher order component (so the params are updated and validated at the top level before fetching new data).

Reply:
Thanks for the insight @sarah-clements  I will look into it and make the changes.